aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -275,33 +275,51 @@ DimVector TensorIterator::invert_perm(IntArrayRef input) const {
   return res;
 }
 
+DimVector TensorIterator::apply_perm_and_mul(IntArrayRef input, int mul) const {
+  TORCH_INTERNAL_ASSERT(!has_coalesced_dimensions_);
+  auto res = DimVector(input.size(), 0);
+  for (size_t i = 0; i < input.size(); i++) {
+    res[i] = input[perm_[i]] * mul;
+  }
+  return res;
+}
+
 void TensorIterator::allocate_outputs() {
   for (int i = 0; i < num_outputs_; i++) {
     auto& op = operands_[i];
     if (!op.tensor.defined()) {
       TORCH_INTERNAL_ASSERT(op.is_type_defined(), "no type for operand", i);
       int element_size = elementSize(op.dtype);
-      op.stride_bytes = compatible_stride(element_size);
-      //check if permutation is just an inverted order
-      bool inverted = true;
-      for (int i = 1; i <= ndim(); i++) {
-        if (perm_[i-1] != ndim() - i) {
-          inverted = false;
-          break;
-        }
-      }
-      auto tensor_shape = invert_perm(shape_);
-      if (inverted) {
-        // can just return contiguous output
-        // it is faster because it avoids allocating 0 size tensor and resizing and restriding it
+      if (requires_channels_last_output_ && ndim() == 4) {
+        auto tensor_shape = invert_perm(shape_);
         op.tensor = at::empty(tensor_shape, op.options());
+        op.tensor.unsafeGetTensorImpl()->empty_tensor_restride(
+            MemoryFormat::ChannelsLast);
+        op.stride_bytes = apply_perm_and_mul(op.tensor.strides(), element_size);
```

Comment:
I cannot convince myself that `op.stride_bytes` and `op.tensor.strides()` are guaranteed to be consistent (one is defined by perm_ which can be whatever, and the other is hardcoded nhwc), and tests are only testing contiguity of output which is guaranteed by `empty_tensor_restride`, not its correctness.

Reply:
Addressed by adding comment